llvm/lib/Transforms/Scalar/AlignmentFromAssumptions.cpp

```diff
@@ -226,6 +223,8 @@ bool AlignmentFromAssumptionsPass::processAssumption(CallInst *ACall,
         LI->setAlignment(NewAlignment);
         ++NumLoadAlignChanged;
       }
+      // The user of a Load uses data - not a pointer!
+      AddUsers = false;
```

Comment:
This fix looks very incomplete to me. We'll no longer follow users of loads, but still for everything else -- for example, what about a call that accepts and returns a pointer?

I believe we should explicitly listing the instructions for which we want to follow pointers, which are bitcasts, geps, selects and phis, and not anything else.

Reply:
I'm not sure I follow. We're working on pointers here, so I would expect that the only type of arithmetic that is relevant here is getelementptr.

Reply:
Oops. Yes, it is a high level IR and GEPs do all with the pointer arithmetic.

Reply:
BTW everything except GEPs makes no sense since the code of the getNewAlignment function calls
`const SCEV *DiffSCEV = SE->getMinusSCEV(PtrSCEV, AASCEV);`
which in order uses the `getPointerBase(LHS) != getPointerBase(RHS)` to compare the pointer bases.
The getBasePointer function only tracks down through the  ADD operation, not PHI or "select" or "bitcast"

Reply:
PR is updated to make algorithm to track the pointer operands users only

Reply:
getPointerBase() also looks through AddRecs, which are phi nodes. But yeah, select is probably not relevant here.